apps/desktop-wallet/src/components/TokenAmountsBox.tsx

```diff
@@ -24,45 +24,62 @@ import { useAppDispatch } from '@/hooks/redux'
 import { links } from '@/utils/links'
 import { openInWebBrowser } from '@/utils/misc'
 
-interface CheckAmountsBoxProps extends AmountBaseProps, BoxProps {
+interface TokenAmountsBoxProps extends AmountBaseProps, BoxProps {
   assetAmounts: AssetAmount[]
+  shouldAddAlphForDust?: boolean
   className?: string
 }
 
-const CheckAmountsBox = ({
+const TokenAmountsBox = ({
   assetAmounts,
   className,
   hasBg,
   hasHorizontalPadding,
   hasVerticalPadding,
   hasBorder,
+  shouldAddAlphForDust,
   ...props
-}: CheckAmountsBoxProps) => {
-  const userSpecifiedAlphAmount = assetAmounts.find((asset) => asset.id === ALPH.id)?.amount
-  const { attoAlphAmount, tokens, extraAlphForDust } = getTransactionAssetAmounts(assetAmounts)
+}: TokenAmountsBoxProps) => {
+  let amounts = assetAmounts.map((asset) => ({
+    ...asset,
+    amount: asset.amount?.toString() ?? '0'
+  }))
 
-  const alphAsset = { id: ALPH.id, amount: attoAlphAmount }
-  const assets = userSpecifiedAlphAmount ? [alphAsset, ...tokens] : [...tokens, alphAsset]
+  let alphForDust = BigInt(0)
+
+  if (shouldAddAlphForDust) {
+    const userSpecifiedAlphAmount = assetAmounts.find((asset) => asset.id === ALPH.id)?.amount
+    const { attoAlphAmount, tokens, extraAlphForDust } = getTransactionAssetAmounts(assetAmounts)
+    const alphAsset = { id: ALPH.id, amount: attoAlphAmount }
+
+    alphForDust = extraAlphForDust
+    amounts = userSpecifiedAlphAmount ? [alphAsset, ...tokens] : [...tokens, alphAsset]
+  }
```

Comment:
Let's put all that into a `useMemo` call, it feels wrong running all this logic on every re-render.

Reply:
Done